src/dbnode/storage/series/buffer.go

```diff
@@ -114,13 +114,15 @@ type bufferStats struct {
 
 type bufferTickResult struct {
 	mergedOutOfOrderBlocks int
-	evictedBuckets         int
+	evictedBucketTimes     []time.Time
```

Comment:
Kinda sucks to allocate millions of tiny arrays temporarily :/

Few options:

1. have both a fixed-size array AND a slice field and use the array field as long as its big enough, otherwise defer to the slice field (which will be a nil pointer that costs us nothing 99.999% of the time)
2. Pass in a slice to the tick method
3. modify the `BlockState` struct to have an extra state field (really gross)

I don't really like any of them, but this seems like a lot of allocations. @prateek thoughts?